### DISCUSSION TITLE
Using pinia-orm in vitest. Cannot instantiate models?

### DISCUSSION THREAD

**Discussion by dschreij** [CodeDredd/pinia-orm#1538] [Q&A] [answered]
Hi there. I have been trying to use pinia-orm in tests running in vitest, but am facing some problems. I used to be able to just use

```ts
const fakeData = new MyModel({
   id: 999
   name: 'Some name'
})
```
in my jest tests using vuex-orm to instantiate models with random data, but somehow in pinia-orm, this returns a model that is undefined. Or at least, it seems to return the type, but it's emtpy? Here is some output of my debug console when inspectig the model object:

<img width="575" alt="image" src="https://github.com/CodeDredd/pinia-orm/assets/689044/80bf2017-f0d8-4e37-9b46-4049e54acc1e">

I also found https://pinia-orm.codedredd.de/api/repository/make that says the repository make function is pretty much the alternative to `new Model()` (but doesn't state if it should be used instead or is interchangeable), but using `useRepo(Organization).make({id: 1, name: 'Some name')` yields the same results as above with `new Model()`. E.g.:

```ts
const pinia = creatTestingPinia()
const repo = useRepo(Organization, pinia)
const randomData = repo.make({...})
// randomdata has type Organization but when trying to access its attributes, it throws the error that and attribute of an undefined object is trying to be read.
```

Can anyone help me to figure this out? Has anyone else successfully used pinia-orm in tests?

**Comment by CodeDredd** [comment]
@dschreij Just look at my tests 😉 
- https://github.com/CodeDredd/pinia-orm/blob/master/packages/pinia-orm/tests/setup.ts
- https://github.com/CodeDredd/pinia-orm/blob/master/packages/pinia-orm/tests/feature/relations/belongs_to_many_retrieve.spec.ts

The big difference betweeen vuex-orm and pinia-orm is, that the complete logic doesn't exist anymore in the store.
Just look at the setup file for my tests.

**Reply by dschreij** [reply]
Thanks @CodeDredd I looked at your tests and it is very enlightening. From what I understand it is not possible to use the models as 'standalone' without initializing a pinia store for each test that uses them? I also use my models as schemas/interfaces to specify the proptypes of a component which doesn't necessarily use the store. I was just to just instantiate a model and fill it with some random data to pass as a prop when testing components that used them, and it seems quote overkill to me to instantiate stores for these tests, just for filling a model with data. 
From how you wrote your tests, I infer that it's not possible to just say `new Model({...fields})` anymore and get a model instance whose field values are set? Or am I overlooking some detail?

**Reply by CodeDredd** [reply, marked as answer]
@dschreij 
yes you can do that. you can just fill the models with the data and how it is set, is defined by the options. Look for example at some of the model unit test: https://github.com/CodeDredd/pinia-orm/blob/master/packages/pinia-orm/tests/unit/model/Model.spec.ts

Maybe i write some helper to mock `useRepo` with return values. I think this will also make testing way easier so that you dont need the store at all.

in your case the best way is to set the models by `new Model({...fields}, { operation: 'get' })`
